packages/client/test/events.spec.ts

```diff
@@ -0,0 +1,188 @@
+      reason: ERROR_REASON,
+      errorCode: ERROR_CODE,
+    });
+  }),
+  events: new OpenFeatureEventEmitter(),
+} as unknown as Provider;
+
+
+
+describe('Events', () => {
+  // set timeouts short for this suite.
+  jest.setTimeout(1000);
+  let API = new (OpenFeatureAPI as any)(); 
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    API = new (OpenFeatureAPI as any)(); //TODO this hacky thing should be removed when https://github.com/open-feature/spec/pull/183/files is implemented
+  });
+
+  describe('Requirement 5.1.1', () => {
+    it('The provider defines a mechanism for signalling the occurrence of an event`PROVIDER_READY`', (done) => {
+      const myProvider: Provider = {
+        metadata: {
+          name: 'mock-events',
+        },
+        initialize: jest.fn(() => {
+          return Promise.resolve(undefined);
+        }),
+        events: new OpenFeatureEventEmitter(),
+      } as unknown as Provider;
+
+      myProvider.events?.addListener(ProviderEvents.Ready, () => {
+        try {
+          expect(API.providerMetadata.name).toBe(myProvider.metadata.name);
+          expect(myProvider.initialize).toHaveBeenCalled();
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+      API.setProvider(myProvider);
+    });
+
+    it('It defines a mechanism for signalling `PROVIDER_CONFIGURATION_CHANGED`', (done) => {
+      API['_apiEvents'] = new OpenFeatureEventEmitter();
+      const myProvider: Provider = {
+        metadata: {
+          name: 'mock-events',
+        },
+        initialize: jest.fn(() => {
+          return Promise.resolve(undefined);
+        }),
+        events: new OpenFeatureEventEmitter(),
+      } as unknown as Provider;
+      API['_apiEvents'].addListener(ApiEvents.ProviderChanged, () => {
+        try {
+          expect(API.providerMetadata.name).toBe(myProvider.metadata.name);
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+      API.setProvider(myProvider);
+
+    });
+
+    it('It defines a mechanism for signalling `PROVIDER_ERROR`', (done) => {
+      //make sure an error event is fired when initialize promise reject
+      ERROR_MOCK_PROVIDER.events?.addListener(ProviderEvents.Error, () => {
+        try {
+          expect(API.providerMetadata.name).toBe(ERROR_MOCK_PROVIDER.metadata.name);
+          expect(ERROR_MOCK_PROVIDER.initialize).toHaveBeenCalled();
+          done();
+        } catch (err) {
+          done(err);
+        }
+      });
+      API.setProvider(ERROR_MOCK_PROVIDER);
+    });
+
+  });
+
+  describe('Requirement 5.2.1, 5.2.3, 5.2.4, 5.2.5 ', () => {
+    it('The `client` MUST provide an `addHandler` function for attaching callbacks to `provider events`, which accepts event type(s) and a `event handler function`.', (done) => {
+      const client = API.getClient();
+      client.addHandler(ProviderEvents.Ready, () => {
+        done();
+      });
+      API.setProvider(MOCK_PROVIDER);
```

Comment:
if this is a dupe (seems to be) you can remove it. It's implicitly tested below.